### PR TITLE
CBO Feb 2024 update

### PR DESCRIPTION
**Pull request by bodiyang**
This PR is to update Tax-Calculator for [the Feb 2024 CBO baseline update](https://www.cbo.gov/publication/59710).

This is also the next step work to [the Tax-Data CBO update](https://github.com/PSLmodels/taxdata/pulls?q=is%3Apr+is%3Aclosed), which update Tax-Calculator with the new files of weights, ratios, growth factors.

The latest budget year will extend to 2034 after the completion of this work.

### CODE REVIEW

File: taxcalc/policy.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ class instance: Policy
     JSON_START_YEAR = 2013  # remains the same unless earlier data added
     LAST_KNOWN_YEAR = 2020  # last year for which indexed param vals are known
```
[the remark this answers is not in the document]
Comment (bodiyang) [reply taxcalc/policy.py]: I've updated it to 2022, which is the last known year of parameter values in `policy_current_law.json` file

**Comment by martinholmer** [comment]
@bodiyang, Thanks for the PR #2729 timing information.

I have two questions:

(1) In the initial comment you make in #2729, you said:
>This is also the next step work to [the Tax-Data CBO update](https://github.com/PSLmodels/taxdata/pulls?q=is%3Apr+is%3Aclosed), which update Tax-Calculator with the new files of weights, ratios, growth factors.

Can you please clarify?   Did you mean "which **updated** Tax-Calculator"?
That is, have the new "weights, ratios, growth factors" from the 2024 CBO projection already been incorporated into the Tax-Calculator master branch?

(2) Exactly how did you generate the new 2026 parameter amounts in the `policy_current_law.json` file included in this pull request?

**Comment by bodiyang** [comment]
> @bodiyang, Thanks for the PR #2729 timing information.
> 
> I have two questions:
> 
> (1) In the initial comment you make in #2729, you said:
> 
> > This is also the next step work to [the Tax-Data CBO update](https://github.com/PSLmodels/taxdata/pulls?q=is%3Apr+is%3Aclosed), which update Tax-Calculator with the new files of weights, ratios, growth factors.
> 
> Can you please clarify? Did you mean "which **updated** Tax-Calculator"? That is, have the new "weights, ratios, growth factors" from the 2024 CBO projection already been incorporated into the Tax-Calculator master branch?
> 
> (2) Exactly how did you generate the new 2026 parameter amounts in the `policy_current_law.json` file included in this pull request?

Hi Martin,

For these two issues:
(1) Not in the master branch yet. It is exactly this PR to update the new weights, ratio, growth factor files.

The CBO update has two steps: First step is to update Tax-Data with the new CBO baseline, where new weights, ratio, growth factor files will be generated; Second step is to update Tax-Calculator with these new files. 
~ This PR is the Second step

(2) The 2026 values in `policy_current_law.json` is generated by running `ppp.py` script.

**Comment by martinholmer** [comment]
In PR #2729 @bodiyang said:
> (1) The CBO update has two steps: First step is to update Tax-Data with the new CBO baseline, where new weights, ratio, growth factor files will be generated; Second step is to update Tax-Calculator with these new files.
~ This PR is the Second step
>
>(2) The 2026 values in policy_current_law.json is generated by running `ppp.py` script.

@bodiyang,  Thanks for the quick answers to my two questions.

But I'm still a little confused about question (1).

Looks like `taxdata` PR #439 was merged into the `taxdata` repository's master branch about two weeks ago.

Have those changes been incorporated into the Tax-Calculator master branch?  
I'm asking because you are using the future tense when you say the "new weights, ratio, growth factor files **will** be generated".

**Comment by bodiyang** [comment]
> In PR #2729 @bodiyang said:
> 
> > (1) The CBO update has two steps: First step is to update Tax-Data with the new CBO baseline, where new weights, ratio, growth factor files will be generated; Second step is to update Tax-Calculator with these new files.
> > ~ This PR is the Second step
> > (2) The 2026 values in policy_current_law.json is generated by running `ppp.py` script.
> 
> @bodiyang, Thanks for the quick answers to my two questions.
> 
> But I'm still a little confused about question (1).
> 
> Looks like `taxdata` PR #439 was merged into the `taxdata` repository's master branch about two weeks ago.
> 
> Have those changes been incorporated into the Tax-Calculator master branch? I'm asking because you are using the future tense when you say the "new weights, ratio, growth factor files **will** be generated".

Sorry about the confusion by future tense. (by mistake, should say "are generated")

This PR is the work to incorporate the files from [Tax-Data PR 439](https://github.com/PSLmodels/taxdata/pull/439) into Tax-Calculator's master branch. 

(Yes, Tax-Data PR 439 was merged two weeks ago; This PR is to incorporate the changes from there)

**Comment by martinholmer** [comment]
In PR #2729, @bodiyang said:
> This PR is the work to incorporate the files from https://github.com/PSLmodels/taxdata/pull/439 into Tax-Calculator's master branch.

OK,  I understand now.   Thanks for clarifying.
Looking forward to this PR being merged very soon.
Thanks for your effort in preparing it.

**Comment by jdebacker** [comment]
@bodiyang Thanks for following up on my review and Martin's questions.  This PR looks good to me.  Merging now.

**Pull request merged**